cpp/include/cudf/strings/detail/utilities.cuh

```diff
@@ -87,6 +91,196 @@ std::unique_ptr<cudf::column> child_offsets_from_string_iterator(
   return make_offsets_child_column(begin, begin + num_strings, stream, mr);
 }
 
+/**
+ * @brief Copies input string data into a buffer and increments the pointer by the number of bytes
+ * copied.
+ *
+ * @param buffer Device buffer to copy to.
+ * @param input Data to copy from.
+ * @param bytes Number of bytes to copy.
+ * @return Pointer to the end of the output buffer after the copy.
+ */
+__device__ inline char* copy_and_increment(char* buffer, const char* input, size_type bytes)
+{
+  memcpy(buffer, input, bytes);
+  return buffer + bytes;
+}
+
+/**
+ * @brief Copies input string data into a buffer and increments the pointer by the number of bytes
+ * copied.
+ *
+ * @param buffer Device buffer to copy to.
+ * @param d_string String to copy.
+ * @return Pointer to the end of the output buffer after the copy.
+ */
+__device__ inline char* copy_string(char* buffer, const string_view& d_string)
+{
+  return copy_and_increment(buffer, d_string.data(), d_string.size_bytes());
+}
+
+/**
+ * @brief Creates child offsets and chars columns by applying the template function that
+ * can be used for computing the output size of each string as well as create the output.
+ *
+ * @tparam SizeAndExecuteFunction Function must accept an index and return a size.
+ *         It must also have members d_offsets and d_chars which are set to
+ *         memory containing the offsets and chars columns during write.
+ *
+ * @param size_and_exec_fn This is called twice. Once for the output size of each string.
+ *        After that, the d_offsets and d_chars are set and this is called again to fill in the
+ *        chars memory.
+ * @param strings_count Number of strings.
+ * @param exec_size Number of rows for executing the `size_and_exec_fn` function.
+ * @param mr Device memory resource used to allocate the returned columns' device memory.
+ * @param stream CUDA stream used for device memory operations and kernel launches.
+ * @return offsets child column and chars child column for a strings column
+ */
+template <typename SizeAndExecuteFunction>
+auto make_strings_children(
+  SizeAndExecuteFunction size_and_exec_fn,
+  size_type strings_count,
+  size_type exec_size,
```

Comment:
Please change the order of `exec_size` to be before `strings_count`, since the other function (`make_strings_children_with_null_mask`) has the `exec_size` parameter comes before `strings_count`.

Reply:
What parameter should be used in the `exclusive_scan` ?

Reply:
Only `exec_size` is used only to execute the template functor. `exclusive_scan` is called to generate the output strings column thus we will use `strings_count`.